controllers/job/reconcilers.go

```diff
@@ -1,32 +1,49 @@
+// Copyright The Snoopy Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package job
 
 import (
 	"context"
 	"fmt"
 
-	jobv1alpha1 "github.com/fennec-project/snoopy-operator/apis/job/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	apimachinery "k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	jobv1alpha1 "github.com/fennec-project/snoopy-operator/apis/job/v1alpha1"
 )
 
 func (r *SnoopyJobReconciler) reconcileCronJobs(snoopyJob *jobv1alpha1.SnoopyJob, cronJobs *batchv1.CronJobList) error {
 
 	for _, cronJob := range cronJobs.Items {
 
-		err := r.Client.Get(context.TODO(), apimachinery.NamespacedName{Namespace: cronJob.ObjectMeta.Namespace, Name: cronJob.ObjectMeta.Name}, &cronJob)
+		cfgCronJob := cronJob
+		err := r.Client.Get(context.TODO(), apimachinery.NamespacedName{Namespace: cronJob.ObjectMeta.Namespace, Name: cronJob.ObjectMeta.Name}, &cfgCronJob)
 		if err != nil {
 			if errors.IsNotFound(err) {
-				err = r.Client.Create(context.Background(), &cronJob)
+				err = r.Client.Create(context.Background(), &cfgCronJob)
 				if err != nil {
 					fmt.Println(err.Error())
 					return err
 				}
```

Comment:
Can we use index instead of creating another variable?
for i, _ := range cronJobs.Items {

     and use &cronJobs[i] here?

}